### ISSUE TITLE
Hangs on opening specific project

### ISSUE THREAD

**Issue by ShonFrazier** [godotengine/godot#42213]
**Godot version:**
3.2.3

**OS/device including version:**
macOS 10.15.6

**Issue description:**
Godot hangs while attempting to open project for editing. Activity Monitor sample seems to indicate it's stuck waiting on a mutex, maybe.

**Steps to reproduce:**
1. Clone https://github.com/DiscoveringGodot/3_Hoppy_Days.git
2. Open the project in the editor.
3. Observe lack of response

**Minimal reproduction project:**
https://github.com/DiscoveringGodot/3_Hoppy_Days.git
Not exactly "minimal" but I can't open the project to begin paring down...

Edit: If I launch Godot from the command line, from within the project folder, it'll play the project. Seems to be a problem with the editor.

**Comment by Calinou** [comment]
I can confirm this on 3.2.3 on Linux.

```text
Godot Engine v3.2.3.stable.official - https://godotengine.org
OpenGL ES 3.0 Renderer: GeForce GTX 1080/PCIe/SSE2
 
ERROR: get_dependencies: Cannot open file 'res://.import/Chiptune_Adventures_1.ogg-52fc949b19c914776132d8c6ca6f4122.oggstr'.
   At: core/io/resource_format_binary.cpp:1047.
ERROR: get_dependencies: Cannot open file 'res://.import/coin_SFX.wav-bc0b68d2c1fd36982fde041f1f1161e5.sample'.
   At: core/io/resource_format_binary.cpp:1047.
ERROR: get_dependencies: Cannot open file 'res://.import/jump1.ogg-8d08467a7f8bcb890557b9f4a378ee0a.oggstr'.
   At: core/io/resource_format_binary.cpp:1047.
ERROR: get_dependencies: Cannot open file 'res://.import/pain.ogg-06a0c2e05ac0a9e26293ec280fc5c195.oggstr'.
   At: core/io/resource_format_binary.cpp:1047.
ERROR: get_dependencies: Cannot open file 'res://.import/Summer_Park.ogg-cf6731f2fc24bf54214f696e58e51c9a.oggstr'.
   At: core/io/resource_format_binary.cpp:1047.
SCRIPT ERROR: GDScript::reload: Parse Error: The function signature doesn't match the parent. Parent signature is: "void update()".
   At: res://Scripts/PlayerAnimation.gd:3.
ERROR: reload: Method failed. Returning: ERR_PARSE_ERROR
   At: modules/gdscript/gdscript.cpp:599.
```

Fixing the script error by renaming the method doesn't resolve the hangup.

**Comment by capnm** [comment]
update() is a function of AnimatedSprite. You have to use a different name ...
otherwise the editor then deadlocks by opening the start scene.

```diff
diff --git a/Scripts/Player.gd b/Scripts/Player.gd
index 57d80ea..8571e8f 100644
--- a/Scripts/Player.gd
+++ b/Scripts/Player.gd
@@ -25,7 +25,7 @@ func _process(delta):
 
 
 func update_animation(motion):
-       $AnimatedSprite.update(motion)
+       $AnimatedSprite.update_sprite(motion)
 
 
 func fall(delta):
diff --git a/Scripts/PlayerAnimation.gd b/Scripts/PlayerAnimation.gd
index 3f50ea3..1757c6f 100644
--- a/Scripts/PlayerAnimation.gd
+++ b/Scripts/PlayerAnimation.gd
@@ -1,6 +1,6 @@
 extends AnimatedSprite
 
-func update(motion):
+func update_sprite(motion):
        if motion.y < 0:
                play("jump")
        elif motion.x > 0:
```

**Comment by ShonFrazier** [comment]
Still hanging after these changes are made.

**Comment by capnm** [comment]
> Still hanging after these changes are made.

It opened the editor by me ;) Try then disable the start scene in `project.godot`

**Comment by ShonFrazier** [comment]
Disabling the start scene does allow it to load without hanging. But in that case, I can leave the original (incorrect) `update(motion)` function names in place and it will load the scene fine when selected from within the editor.

It comes down to this change to the project to prevent hanging:
```diff --git a/project.godot b/project.godot
index d7fc8f7..aafba7a 100644
--- a/project.godot
+++ b/project.godot
@@ -11,7 +11,7 @@ config_version=3
 [application]
 
 config/name="2_Hoppy_Days"
-run/main_scene="res://Scenes/Levels/Level1.tscn"
+#run/main_scene="res://Scenes/Levels/Level1.tscn"
 config/icon="res://icon.png"
 
 [autoload]
```

**Comment by capnm** [comment]
> The editor probably shouldn't hang regardless.

```
-run/main_scene="res://Scenes/Levels/Level1.tscn"
+;run/main_scene="res://Scenes/Levels/Level1.tscn"
```

In fact, when I activate the start scene and close the editor, it then hangs again :-(
Unfortunately, the editor has many such issues. It uses intertwined threads for updating that are difficult to track and debug (at least for me;) You can try disabling parts of your game and find out what is causing this.

**Comment by capnm** [comment]
FYI disabling the thumbnail editor feature as described in #42785, also fixed for me your project.

**Comment by akien-mga** [comment]
Closing as duplicate of #42785 (same issue and this one was reported first, but #42785 has more extensive debug findings).
